src/mgmt/srv_drpc.c

```diff
@@ -747,6 +747,136 @@ ds_mgmt_drpc_pool_get_acl(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	free_ace_list(ace_list, ace_nr);
 }
 
+/* Convert d_rank_list_t values to a string of comma-separated ranks.
+ * Allocates, fills and returns string (to be freed by caller).
+ * If unsuccessful, returns NULL.
+ */
+static char *
+rank_list_to_csvstr(d_rank_list_t *rl)
+{
+	char	*buf = NULL;
+	int	 buflen = 16;	/* grow as needed */
+	int	 bufidx;
+	int	 i;
+
+	D_ALLOC(buf, buflen);
+	if (buf == NULL)
+		goto out;
+
+	bufidx = sprintf(buf, "%u", rl->rl_ranks[0]);
+	for (i = 1; i < rl->rl_nr; i++) {
+		bufidx += snprintf(&buf[bufidx], buflen-bufidx,
+				      ",%u", rl->rl_ranks[i]);
+		if (bufidx >= buflen) {
+			char *extra = NULL;
+
+			buflen *= 2;
+			D_ALLOC(extra, buflen);
```

Comment:
It occurs to me there is a D_REALLOC - would that be better suited to use here?

Reply:
I started out in that direction here, and ended up using the same pattern as found in ds_mgmt_drpc_pool_create(). I think the issue is the following sprintf needs to fill the expanded (or possibly new) allocation with the original string content. There is an opportunity to have ds_mgmt_drpc_pool_create() also call this new utility function.